File: Python3/2144. Minimum Cost of Buying Candies With Discount.py
class Solution:
    def minimumCost(self, cost: List[int]) -> int:
        if len(cost) < 3:
            return sum(cost)

        cost.sort(reverse=True)
        ans = 0

        for i, c in enumerate(cost):
            if (i + 1) % 3:
                ans += cost[i]

        return ans

File: Java/2144. Minimum Cost of Buying Candies With Discount.java
class Solution {
    public int minimumCost(int[] cost) {
        int ans = 0, N = cost.length - 1;
        Arrays.sort(cost);

        for (int i = N; i >= 0; i--) {
            if ((N - i) % 3 != 2) {
                ans += cost[i];
            }
        }

        return ans;
    }
}